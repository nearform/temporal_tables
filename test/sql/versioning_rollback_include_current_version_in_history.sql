-- Rollback Behavior Test
CREATE TABLE versioning_rollback (a bigint, "b b" date, sys_period tstzrange);

-- Insert initial data.
INSERT INTO versioning_rollback (a, "b b", sys_period) VALUES (1, '2020-01-01', tstzrange('2000-01-01', NULL));

CREATE TABLE versioning_rollback_history (a bigint, "b b" date, sys_period tstzrange);

CREATE TRIGGER versioning_rollback_trigger
BEFORE INSERT OR UPDATE OR DELETE ON versioning_rollback
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', 'versioning_rollback_history', false, false, true);

-- Test rollback during update.
BEGIN;

UPDATE versioning_rollback SET a = 2 WHERE a = 1;

-- Rollback the transaction.
ROLLBACK;

-- Ensure no history record was created.
SELECT * FROM versioning_rollback_history;  -- Expecting 0 rows in history.

-- Ensure original data is unchanged.
SELECT * FROM versioning_rollback;  -- Expecting original value a = 1.

-- Cleanup
DROP TABLE versioning_rollback;
DROP TABLE versioning_rollback_history;
