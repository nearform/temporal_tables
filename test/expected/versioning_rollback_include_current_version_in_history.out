-- Rollback Behavior Test
CREATE TABLE versioning_rollback (a bigint, "b b" date, sys_period tstzrange);
-- Insert initial data.
INSERT INTO versioning_rollback (a, "b b", sys_period) VALUES (1, '2020-01-01', tstzrange('2000-01-01', NULL));
CREATE TABLE versioning_rollback_history (a bigint, "b b" date, sys_period tstzrange);
CREATE TRIGGER versioning_rollback_trigger
BEFORE INSERT OR UPDATE OR DELETE ON versioning_rollback
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', 'versioning_rollback_history', false, false, true);
-- Test rollback during update.
BEGIN;
UPDATE versioning_rollback SET a = 2 WHERE a = 1;
-- Rollback the transaction.
ROLLBACK;
-- Ensure no history record was created.
SELECT * FROM versioning_rollback_history;  -- Expecting 0 rows in history.
 a | b b | sys_period 
---+-----+------------
(0 rows)

-- Ensure original data is unchanged.
SELECT * FROM versioning_rollback;  -- Expecting original value a = 1.
 a |    b b     |         sys_period          
---+------------+-----------------------------------
 1 | 01-01-2020 | ["Sat Jan 01 00:00:00 2000 UTC",)
(1 row)

-- Cleanup
DROP TABLE versioning_rollback;
DROP TABLE versioning_rollback_history;
